### PR TITLE
[SelectionDAG][X86] Preserve unpredictable metadata for conditional branches in SelectionDAG, as well as JCCs generated by X86 backend.

### PR DESCRIPTION
**Pull request by tianqingw**
This builds on 09515f2c2, which preserves unpredictable metadata in CodeGen for `select`. This patch does it for conditional branches.

### CODE REVIEW

**Comment by DianQK** [comment]
Sorry, I cannot review this PR. :)

File: llvm/test/CodeGen/X86/unpredictable-brcond.ll [hunk 1]
```diff
@@ -0,0 +1,122 @@
+; NOTE: Assertions have been autogenerated by utils/update_mir_test_checks.py UTC_ARGS: --version 5
```
Comment (phoebewang) [inline llvm/test/CodeGen/X86/unpredictable-brcond.ll:1]: Pre-commit the test case?

File: llvm/include/llvm/CodeGen/SwitchLoweringUtils.h [hunk 2]
```diff
@@ -137,29 +137,34 @@ struct CaseBlock {
   SDLoc DL;
   DebugLoc DbgLoc;
 
-  // Branch weights.
+  // Branch weights and predictability.
   BranchProbability TrueProb, FalseProb;
+  bool IsUnpredictable;
```
Comment (phoebewang) [inline llvm/include/llvm/CodeGen/SwitchLoweringUtils.h:142]: Where this get used?

File: llvm/test/CodeGen/X86/unpredictable-brcond.ll (hunk 1 above)
Comment (RKSimon) [reply llvm/test/CodeGen/X86/unpredictable-brcond.ll:1]: +1

File: llvm/include/llvm/CodeGen/SwitchLoweringUtils.h (hunk 2 above)
Comment (KanRobert) [reply llvm/include/llvm/CodeGen/SwitchLoweringUtils.h:142]: +1

Comment (KanRobert) [reply llvm/include/llvm/CodeGen/SwitchLoweringUtils.h:142]: Never mind. I found https://github.com/llvm/llvm-project/commit/09515f2c20111628ce81ad5f40e12e5f6af5ed2f in your description.

File: llvm/test/CodeGen/X86/unpredictable-brcond.ll (hunk 1 above)
Comment (tianqingw) [reply llvm/test/CodeGen/X86/unpredictable-brcond.ll:1]: Good idea. Submitted #102262.

File: llvm/include/llvm/CodeGen/SwitchLoweringUtils.h (hunk 2 above)
Comment (tianqingw) [reply llvm/include/llvm/CodeGen/SwitchLoweringUtils.h:142]: See `SelectionDAGBuilder::visitSwitchCase`.

**Review by phoebewang — APPROVED** [review]
LGTM.

**Review by KanRobert — APPROVED** [review]
LGTM

**Pull request merged**